includes/admin/upgrades/upgrade-functions.php

```diff
@@ -1328,499 +1328,294 @@ function edd_include_file_download_log_migration_batch_processor( $class = '' )
-	<script>
-		jQuery( function($) {
-			$(document).ready(function () {
-
-				function processNext( current ) {
-					current.find( '.edd-migration.allowed button' ).prop( 'disabled', true );
-					current.find('.dashicons.dashicons-yes').show();
-					current.find('.message.initial').hide();
-					current.find('.message.complete').show();
-
-					var auto_start_next_step = true;
-
-					if ( current.find('.edd-new-count') ) {
-						var new_count = current.find('.edd-new-count').text(),
-							old_count = current.find('.edd-old-count').text();
-
-						auto_start_next_step = new_count === old_count;
-					}
-
-					var next_step_wrapper = current.next();
+	<p>
+		<!-- @todo wording -->
+		<?php esc_html_e( 'Easy Digital Downloads needs to perform upgrades to your WordPress database. Your store data will be migrated to custom database tables to improve performance and efficiency. This process may take a while.', 'easy-digital-downloads' ); ?>
+		<strong><?php esc_html_e( 'Please create a full backup of your website before proceeding.', 'easy-digital-downloads' ); ?></strong>
+	</p>
 
-					if ( next_step_wrapper.hasClass( 'metabox-holder' ) ) {
-						if ( next_step_wrapper.find( '.edd-migration.allowed' ).length ) {
-							next_step_wrapper.find('.edd-migration.allowed').show();
-							next_step_wrapper.find('.edd-migration.unavailable').hide();
+	<p>
+		<?php printf( esc_html__( 'This migration can also be run via WP-CLI with the following command: %s. This is the recommended method for large sites.', 'easy-digital-downloads' ), '<code>wp edd v30_migration</code>' ); ?>
+	</p>
 
-							if (auto_start_next_step) {
-								next_step_wrapper.find('.edd-export-form:not(#edd-remove-legacy-data-form)').submit();
-							}
+	<?php
+	// Only show the migration form if there are still upgrades to do.
+	if ( ! $can_remove_legacy_data ) : ?>
+	<form id="edd-v3-migration" class="edd-v3-migration" method="POST">
+		<p>
+			<label for="edd-v3-migration-confirmation">
+				<input type="checkbox" id="edd-v3-migration-confirmation" class="edd-v3-migration-confirmation" name="backup_confirmation" value="1">
+				<?php esc_html_e( 'I have secured a backup of my website data.', 'easy-digital-downloads' ); ?>
+			</label>
+		</p>
+		<input type="hidden" name="_wpnonce" value="<?php echo esc_attr( wp_create_nonce( 'edd_process_v3_upgrade' ) ); ?>">
+		<button type="submit" id="edd-v3-migration-button" class="button button-primary disabled" disabled>
+			<?php esc_html_e( 'Upgrade Easy Digital Downloads', 'easy-digital-downloads' ); ?>
+		</button>
+		<div class="edd-v3-migration-error edd-hidden"></div>
+	</form>
+	<?php endif
+
+	/*
+	 * Progress is only shown immediately if the upgrade is in progress. Otherwise it's hidden by default
+	 * and only revealed via JavaScript after the process has started.
+	 */
+	?>
+	<div id="edd-migration-progress" <?php echo count( array_filter( $upgrade_statuses ) ) ? '' : 'class="edd-hidden"'; ?>>
+		<ul>
```

Comment:
I think this could reasonably be changed to an ordered list, although I'm not sure it matters significantly--since the WP admin does show the numbers, but not standard bullets. I was just thinking it might be nice to show the information such as "Step 1 of 8" and so on.

Reply:
I think I'm going to leave it. I don't think it's hugely important one way or another, but adding numbers will add a bit more "business" and ultimately the spirit of this PR is to keep it really simple.